### ISSUE TITLE
Entry `builddir: <dir>` silently dropped from `cabal.project`

### ISSUE THREAD

**Issue by andreasabel** [haskell/cabal#7941]
Entries `builddir: <dir>` are silently dropped from `cabal.project`.  (Context: https://github.com/haskell/cabal/issues/7368#issuecomment-1025174066)

Whether they should be supported or not is discussed in #5271.

In the meantime, while that issue cooks, `cabal` should give a warning that `builddir:` entries are ignored/unknown/unsupported.

**Comment by ulysses4ever** [comment]
Historical reference: the `--builddir` option (and the docs saying that it won't work from inside `cabal.project`)  comes from #4124 where the description reads:

> It's not too easy to make the value of 'dist-dir' (_currently `builddir` -- u4e_) saveable
> to the cabal.project file, as we'd prefer to load directly
> from the cache if cabal.project is not updated.

I should confess I don't understand what it means though.

**Comment by Kleidukos** [comment]
@mixphix didn't you stumble on this bug by any chance?

**Comment by ulysses4ever** [comment]
@Kleidukos you may be thinking of https://github.com/haskell/cabal/issues/8947

**Comment by mixphix** [comment]
Yeah, which I learned was a duplicate of #5271. It turns out that HLS puts things into `dist-newstyle` anyway, so I've given up on the tractability of what I was trying to do there.
